### PR TITLE
feat: add sidebar

### PR DESCRIPTION
**Pull request by neverwanalive**

### CODE REVIEW

File: src/lib/theme/animations.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { Keyframes, keyframes } from 'styled-components'
+
+export const show = (direction: string): Keyframes => keyframes`
```
Comment (ordinaryask) [inline src/lib/theme/animations.ts]: prettier on

Comment (ordinaryask) [inline src/lib/theme/animations.ts]: maybe use second arg instead of the same animation

File: src/lib/core/Sidebar/Sidebar.types.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { ReactNode } from 'react'
+
+import { Color, ColorWithOpacity, Position } from '../../theme'
+
+export interface SidebarProps extends React.HTMLAttributes<HTMLElement> {
+  isOpen: boolean
+  sidebarDirection: Position
```
Comment (ordinaryask) [inline src/lib/core/Sidebar/Sidebar.types.ts]: better use here 'start' | 'end'

Comment (ordinaryask) [inline src/lib/core/Sidebar/Sidebar.types.ts]: and rename to direction

File: src/lib/theme/animations.ts (hunk 1 above)
Comment (ordinaryask) [inline src/lib/theme/animations.ts]: i guess "direction" is wrong word for this arg

Comment (ordinaryask) [reply src/lib/theme/animations.ts]: like first arg will be start, second end
0 15.625rem
-15.625rem 0

File: src/lib/core/Sidebar/Sidebar.component.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  onClose,
+  crossColor = Color.GRAY_100,
+  ...otherProps
+}) => {
+  const stopPropogation = (e: SyntheticEvent): void => {
+    e.stopPropagation()
+  }
+
+  return (
+    <Wrapper
```
Comment (ordinaryask) [inline src/lib/core/Sidebar/Sidebar.component.tsx]: is isOpen=false, sidebar should be closed, but unfortunately we see wrapper

**Review by ordinaryask — CHANGES_REQUESTED** [review]

File: src/lib/theme/animations.ts (hunk 1 above)
Comment (neverwanalive) [reply src/lib/theme/animations.ts]: не ебет, не работает

**Review by ordinaryask — APPROVED** [review]

**Pull request merged**